Add more exempt tags for stalebot; Make stalebot to close PRs/issues in 28 days rather than in 7/14 days

leventov opened the pull request:
See [the discussion in the mailing list](https://lists.apache.org/thread.html/70c4ffb473f48033550575da565962a85b0e74795aef7b328bbe9d91@%3Cdev.druid.apache.org%3E).

### Description

#### Stalebot exemption labels
This PR greatly extends the list of labels which are exempt for stalebot. The only labels which are *not* exempt are:

- `Backport`
- `Difficulty - *`: not used these days, old issues should be revised
- `Discuss`
- `duplicate`
- `Feature/Change Description`: no consensus has been reached regarding this label in the discussion in the mailing list
- `Feature`: no consensus has been reached regarding this label in the discussion in the mailing list
- `Flaky test`: tests tend to become less flaky in time, so these issues should be revised
- `invalid`
- `Priority - *`: not used these days, old issues should be revised
- `Question`
- `Release Notes`
- `WIP`
- `wontfix`

#### Stalebot closing timeout

I've changed the timeout to 28 days for both PRs and issues. I don't see why there should be a difference between PRs and issues. 14 days feels like a too short period. Stale issues and PRs don't block anybody, so there should be no hurry with closing them.

#### Added `Uncategorized problem report` label in "Problem report" template

#### Use "label" instead of "tag" term in `committer-instructions.md`

"Label" is the official Github term.

<hr>

This PR has:
- [x] been self-reviewed.

gianm commented:
What's the rationale behind adding this set of exempt labels? (I don't think this was well explored in the linked thread.) I think it's likely that at least one of these labels would be set by a committer on pretty much every issue and PR they touch (especially the "Area" labels but also ones like "Improvement"), which IMO defeats the purpose of the stalebot for PRs. I don't think we really want a PR to stay open forever just because someone gave it an "Area" label, but then in the end it lost momentum.

If the idea is to keep _issues_ open forever with these labels (i.e. keep properly categorized issues open forever) then it would be better to exempt these labels for issues only (not PRs).

The closing timeout changes (to 4 weeks) sound good to me.

The committer-instructions changes look good too -- thanks.

leventov commented:
I overlooked the PR/issue distinction. Indeed, I also don't see the point of having exempt labels for PRs - even `Bug`/`Security`/`Proposal` which used to be exempt for PRs before. I've updated the PR. I've also added `Uncategorized problem report` label discussed in the thread in the mailing list to "Problem report" template.

clintropolis commented:
It might be worth going over the set of labels to consider if there are any that we should consider removing. For example, I think the `Difficulty - *` and `Priority - *` labels could be purged. I'll try to find time to have a look at them and see if there are any others I think could be dropped and raise a dev list thread about it.
